### PR TITLE
This Time Really Support Laravel11

### PR DESCRIPTION
**Pull request by fuwasegu**
Since the original version did not follow Laravel 11 properly, I'm going to fork it and follow it on my own.

Basically, we restored the dependency on DBAL in the library that was broken by Laravel11, and restored the code that was removed from Laravel.

The following major changes have been made
- Laravel11 or higher required
- PHP8.3 or higher required
- PHPStan is installed
- PHP-CS-Fixer is installed

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /vendor
 /.idea
-.ecs_cache
 phpunit.xml
 .phpunit.result.cache
 /build
 composer.lock
+/.cache
+/.phpunit.cache
```
Comment (mpyw) [inline .gitignore]:
```suggestion
/.phpunit.cache

```

File: .php-cs-fixer.php [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            ->setRiskyAllowed(true);
+    }
+}
+
+return (new PhpCsFixerConfig())
+    ->setCacheFile(__DIR__ . './cache/php-cs-fixer.json')
+    ->setFinder(
+        (new Finder())
+            ->in(__DIR__),
+    );
```
Comment (mpyw) [inline .php-cs-fixer.php]:
```suggestion
    );

```

File: src/PostgresDriver.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        array $driverOptions = [],
+    ): PDOConnection {
+        $pdo = $params['pdo'] ?? null;
+
+        if (!$pdo instanceof PDO) {
+            throw new InvalidArgumentException('Laravel requires the "pdo" property to be set and be a PDO instance.');
+        }
+
+        return new PDOConnection($pdo);
+    }
```
Comment (mpyw) [inline src/PostgresDriver.php:28]: Is this `$pdo` already resolved (`Closure` unwrapped)?

File: src/PostgresExtensionsServiceProvider.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+declare(strict_types=1);
+
+namespace Fuwasegu\Postgres;
+
+use Fuwasegu\Postgres\Connectors\ConnectionFactory;
+use Illuminate\Database\DatabaseServiceProvider;
+use Illuminate\Database\DatabaseTransactionsManager;
+use Override;
+
+class PostgresExtensionsServiceProvider extends DatabaseServiceProvider
```
Comment (mpyw) [inline src/PostgresExtensionsServiceProvider.php]: Is this `extends` required?

File: src/PostgresDriver.php (hunk 3 above)
Comment (mpyw) [reply src/PostgresDriver.php:28]: Confirmed

File: src/PostgresExtensionsServiceProvider.php (hunk 4 above)
Comment (fuwasegu) [reply src/PostgresExtensionsServiceProvider.php]: [f987a13](https://github.com/fuwasegu/laravel-pg-extensions-next/pull/2/commits/f987a132f716e6c0a3f9df3c6e48328785254cc3)

**Review by mpyw — APPROVED** [review]

**Pull request merged**